module3/homework/advanced_calculator/advancedCalculator.cpp

```diff
@@ -0,0 +1,70 @@
+#include "advancedCalculator.hpp"
+
+#include <cmath>
+#include <functional>
+#include <map>
+#include <regex>
+
+const std::map<std::string, std::function<double(double, double)>> operationsMap{
+    {"+", std::plus<double>()},
+    {"-", std::minus<double>()},
+    {"*", std::multiplies<double>()},
+    {"/", std::divides<double>()},
+    {"%", std::modulus<int>()},
+    {"!", [](double number, double zero) { return (number <= 0 ? 1.0 : tgamma(number + 1.0)); }},
+    {"^", [](double base, double exponent) { return pow(base, exponent); }},
+    {"$", [](double num, double root) { return pow(num, 1.0 / root); }}};
+
+bool checkCharacters(std::string input) {
+    std::string allowedSigns{" ,.!+/%$-+*^"};
```

Comment:
I think it should be const and would consider moving it to the header.

Reply:
+1